integration/spark/src/main/spark3/java/io/openlineage/spark3/agent/lifecycle/plan/DatasetSourceVisitor.java

```diff
@@ -21,35 +27,75 @@
  */
 public class DatasetSourceVisitor extends QueryPlanVisitor<LogicalPlan, OpenLineage.Dataset> {
 
+  private static final String ICEBERG = "iceberg";
+
   @Override
-  public boolean isDefinedAt(LogicalPlan x) {
-    return findDatasetSource(x).isPresent();
+  public boolean isDefinedAt(LogicalPlan logicalPlan) {
+    return findDatasetProvider(logicalPlan).equals(ICEBERG)
+        || findDefaultDatasetSource(logicalPlan).isPresent();
   }
 
-  private Optional<DatasetSource> findDatasetSource(LogicalPlan plan) {
+  private Optional<DatasetSource> findDefaultDatasetSource(LogicalPlan plan) {
     if (plan instanceof LogicalRelation) {
       if (((LogicalRelation) plan).relation() instanceof DatasetSource) {
         return Optional.of((DatasetSource) ((LogicalRelation) plan).relation());
       }
     } else if (plan instanceof DataSourceV2Relation) {
       DataSourceV2Relation relation = (DataSourceV2Relation) plan;
+
       if (relation.table() instanceof DatasetSource) {
         return Optional.of((DatasetSource) relation.table());
       }
     }
-
     return Optional.empty();
   }
 
+  private String findDatasetProvider(LogicalPlan plan) {
+    return Optional.of(plan)
+        .filter(x -> x instanceof DataSourceV2Relation)
+        .map(x -> (DataSourceV2Relation) x)
+        .map(DataSourceV2Relation::table)
+        .map(Table::properties)
+        .map(properties -> properties.get("provider"))
+        .map(String::toLowerCase)
+        .orElse("unknown");
+  }
+
+  private OpenLineage.Dataset findDatasetForIceberg(DataSourceV2Relation relation) {
+    Map<String, String> properties = relation.table().properties();
+
+    String namespace = properties.getOrDefault("location", "unknown");
+    namespace = namespace.startsWith("/") ? "file:/" + namespace : namespace;
```

Comment:
that should be `"file://" + namespace` - otherwise, `"file:" + namespace`. An absolute file URL should start with 3 slashes or 1. `file://` is the scheme, meaning everything after the second slash is the path. If the path doesn't start with a slash, that would indicate a relative path, not an absolute one.

Reply:
OK.